Dead station

SergeiKoralev opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

DeadStation is designed for exactly what the name implies, deadpop. It's focused around drawing people as quickly as possible into central/core areas of the station and making sure that players are able to see one another. On top of that there is a whole ton of maintenance space for you all to do evil things to one another. Primary features include... 
* multi-deck configuration that was included in design, meaning it shouldn't feel haphazard or ramshackle mechanically.
* A unique supermatter engine configuration
* Service-centric layout, encouraging passage around or through the bar/kitchen.
* Centrally located science department to encourage additional traffic in core areas of the station.
* Upper deck usage for ancillary department areas such as xenobiology, toxins, nanite lab, and chemical factory.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

There are low-population maps, but for the most part they don't really focus on encouraging traffic around the station so you can see other players. In theory this should help with that. It also follows a more traditional footprint than Kilo or Pubby does. It also adds a true designed and built multi-z map, providing additional diversity from existing maps. Extensive maintenance space also provides plenty of areas for crime to happen even with a design that encourages player interaction.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>


![2022 05 14-12 46 01](https://user-images.githubusercontent.com/43328793/168442983-0da32720-e3a5-4266-9d8e-b779c8c7c8d8.png)
![2022 05 14-12 45 43](https://user-images.githubusercontent.com/43328793/168442985-4b7b474f-cc15-4f58-a721-1020fc3ef99d.png)

</details>

## Changelog
:cl:
add: Adds a new station (Deadstation)
/:cl:

SergeiKoralev committed:
it's a satrt?

SergeiKoralev committed:
progress?

SergeiKoralev committed:
probably the first revert point.

SergeiKoralev committed:
congress.

SergeiKoralev committed:
Engine

SergeiKoralev committed:
Atmosia

SergeiKoralev committed:
Filled out medical and service some.

SergeiKoralev committed:
cargo, xenobio

SergeiKoralev committed:
e

SergeiKoralev committed:
cap, hop, engi

SergeiKoralev committed:
engi

SergeiKoralev committed:
multi-z initilization

SergeiKoralev committed:
Starting to apply areas.

SergeiKoralev committed:
more areas

SergeiKoralev committed:
more areas and offices

SergeiKoralev committed:
air alarms and lights in engi

SergeiKoralev committed:
wall mounted items have directional variants now for mappers.

SergeiKoralev committed:
wall fixtures.

SergeiKoralev committed:
security work

SergeiKoralev committed:
more sec work

SergeiKoralev committed:
upper deck work

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
hate? Hate.

SergeiKoralev committed:
hallways and maint areas

SergeiKoralev committed:
AI sat, Tcomms, conference room, bridge

SergeiKoralev committed:
reeeee

SergeiKoralev committed:
It loads now.

SergeiKoralev committed:
solars!

SergeiKoralev committed:
fuck multi-z

SergeiKoralev committed:
fixes

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
Fixes multi-z. Continues the work.

SergeiKoralev committed:
More progress

SergeiKoralev committed:
Toxins

SergeiKoralev committed:
I forgot an RD server.

SergeiKoralev committed:
Sauna!

SergeiKoralev committed:
More areas.

SergeiKoralev committed:
Powernet troubleshooting

SergeiKoralev committed:
More Wires.

SergeiKoralev committed:
work continues

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
more changes

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
Request Consoles, lower deck.

SergeiKoralev committed:
brings in multiz power adapter from NSV13

SergeiKoralev committed:
finished RCs, finished intercoms.

SergeiKoralev committed:
starts on disposals

SergeiKoralev committed:
more tweaks and bug fixes. Probably ready for a live test.

SergeiKoralev committed:
Disposals done-ish

SergeiKoralev committed:
More minor tweaks.

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
removes my multiz fixes because Corvid fixed them.

SergeiKoralev committed:
6 layer pipes were a scary thing. Over it now.

SergeiKoralev committed:
more corrections to the map post changes

SergeiKoralev committed:
It's back to where it was.

SergeiKoralev commented:
Alright, just to be clear, the current status of this is that it is ready for a play-test. It is, currently, as complete as I can make it on my own without in-depth understanding of every job (or the patience to test it all myself).

mcmeiler commented on _maps/deadstation.json:
```diff
@@ -0,0 +1,12 @@
+	    "whiteship": "whiteship_meta"
+    },
+	"traits": [{"Up" : 1, "Linkage": "Self"}, {"Down": -1, "Baseturf" : "/turf/open/openspace", "Linkage": "Self"}]
+}
```
Newline
```suggestion
}

```

SergeiKoralev committed:
Update _maps/deadstation.json

Co-authored-by: mcmeiler <83819203+mcmeiler@users.noreply.github.com>

MNarath1 commented:
> _maps/map_files/DeadStation/Deadstation.dmm:10963:error: invalid variable name: harm_intent_damage
>WARNING: Found potential invalid file reference to _maps/map_files/Deadstation/Deadstation.dmm in _maps/_maps/deadstation.json
>WARNING: Found potential invalid file reference to _maps/map_files/Deadstation/Deadstation2.dmm in _maps/_maps/deadstation.json

SergeiKoralev committed:
replaces HOS' spider, which was actually a bat, with a goose.

MNarath1 commented:
Eh its actually normal for the hos spider to be a bat juust remove the single var on it that is long not a thing anymore but somehow still keeps showing up

MNarath1 commented:
En short harm intend damage

AnCopper commented:
![image](https://user-images.githubusercontent.com/82319489/168448995-065b3435-24e9-4a5a-97bf-658aed57b0cc.png)
![image](https://user-images.githubusercontent.com/82319489/168448998-ab92142d-8120-4bc5-8730-0f3a0366a074.png)

SergeiKoralev commented:
> ![image](https://user-images.githubusercontent.com/82319489/168448995-065b3435-24e9-4a5a-97bf-658aed57b0cc.png) ![image](https://user-images.githubusercontent.com/82319489/168448998-ab92142d-8120-4bc5-8730-0f3a0366a074.png)

there were already pictures in the pictures and video dropdown of the PR.

Archanial commented:
With our current map rotation we can't have this enabled.

And if we can't have this enabled what's the point of having the map?

williamcll commented:
Multi-Z wires still don't work, power on the lower deck can't be transferred to the upper deck yet.

SergeiKoralev commented:
> With our current map rotation we can't have this enabled.

Why can't we?



> Multi-Z wires still don't work, power on the lower deck can't be transferred to the upper deck yet.

They work. I tested them.

williamcll commented:
> > With our current map rotation we can't have this enabled.
> 
> Why can't we?
> 
> > Multi-Z wires still don't work, power on the lower deck can't be transferred to the upper deck yet.
> 
> They work. I tested them.

I think he means that there's too many maps on rotation? Because Tubestation and Corgstation are being merged first.

PowerfulBacon commented:
We have a lot of map PRs right now, so this might take some time to get through

PowerfulBacon commented on config/maps.txt:
```diff
@@ -41,6 +41,14 @@ map deltastation
+	votable
+endmap
+
+#Does not support fastmos
```
Any point for this message?

SergeiKoralev replied:
Honestly, unless a fastmos PR comes along there isn't any point to that. If you want me to remove it I can.

PowerfulBacon replied:
Yea, probably best since all our maps use the new firelocks if that's what you mean by does not support fastmos

SergeiKoralev replied:
removed in 5aed385

SergeiKoralev committed:
removes the fastmos comment from the config.

PestoVerde322 commented:
Ok I've checked the map, and first thing that was very evident is that the Z-layer is in two different maps, while this is a system used in other places, Beestation can support maps with Z layers, you can do this by using StrongDMM and set the map size to 2Z (and also use StrongDMM2, which has proper multi-Z support).

Second off, this:

![immagine](https://user-images.githubusercontent.com/75247747/170230953-d2ce78f7-4890-45de-9b4f-26142c5fd4b0.png)

I suppose this is a leftover from the FastMos airlock system, but really having two firelocks like this is nothing but redundant, I strongly suggest to do as any other common map do, which is to shove the firelock under the airlocks.

![immagine](https://user-images.githubusercontent.com/75247747/170231360-08261016-3f4c-4a5e-8b0b-8bfbe2dc1820.png)

Any Virology trash will hit the wall over there, not so biohazard safe I'd say.

![immagine](https://user-images.githubusercontent.com/75247747/170231707-a2cba157-f23a-426f-8903-79da3bf5c19d.png)

All the assistant spawns are localized within the bar, suggest to move them around really otherwise not only it will feel unnatural, but in the (rare) case if this map suddenly goes highpop, and many people decide to sign as an assistant, there's basically a Battle royale light to get all the goodies of the tech storage.

![immagine](https://user-images.githubusercontent.com/75247747/170232395-6271b673-aa14-42a0-9869-8da75668a57c.png)

This maint "corridor" feels weird, suggest to actually use a legit corridor and distribute the maint rooms around (or to remove them altogether, considering the station is on multi-z)

![immagine](https://user-images.githubusercontent.com/75247747/170232550-3768ea96-b543-4ada-a2be-b3138558bd8b.png)

also those APCs around maint will act weird with wiring.

![immagine](https://user-images.githubusercontent.com/75247747/170232996-4492d057-081d-4ad2-8062-c645ce277a1d.png)

the Gravity generator could be relocated somewhere else, where it is right now it can be prime target for nuke ops.

This comes down a bit more to personal taste, but a lot of those empty gaps kinda feel... pointless in a station environment, where they could be filled instead with other stuff, and also a lot of ladders are just weirdly located:

![immagine](https://user-images.githubusercontent.com/75247747/170233755-b9417a1b-dc6f-4e0d-9d8e-9090f7806b83.png)

![immagine](https://user-images.githubusercontent.com/75247747/170233793-69992cec-a13a-477a-b797-4e009e86fb47.png)

you could place some stairs instead and some of the rooms, like the tech storage (which currently sits in a quite isolated location next to the server room, which by the way it's WAY too big for any station) and, speaking of, i am not 100% sure how mechs would get up or down a floor in this station considering they can't use stairs.

![immagine](https://user-images.githubusercontent.com/75247747/170234155-d4cef75f-414d-499a-9db2-ab3ebb5af1ed.png)

Telecomms is quite thick, but those internal double walls don't really make much sense if you can just breach a window up there and get in easly, and speaking of double walls:

![immagine](https://user-images.githubusercontent.com/75247747/170234421-ef64335a-ccdc-4424-858e-9d88474383c2.png)

avoid those settings, fill them with maint junk instead, and really, there's an overabundance of random maint in the second floor, consider trimming them down a little or add some more connected rooms.

![immagine](https://user-images.githubusercontent.com/75247747/170234630-c99c7cc9-4d0f-4f1d-85e3-c40c63ee68fe.png)

Perkele, a sauna center!

Also many cables seems to use an invalid icon state, Suggest to just use an existing map for reference on cables.

![immagine](https://user-images.githubusercontent.com/75247747/170235096-9da01e06-1ecc-4f25-b4bb-d311383a6229.png)


Overall this station might need plenty of tweaks, but it might be interesting to see a tight map with lots of maint rooms.
Also update your game version, the most recent version i got is quite dated.

SergeiKoralev commented:
Alright, responding now, fixing later. 

- Differentiation in map files is intentional and there is realistically no significant difference. I also couldn't get a single .dmm with Z2 to work, so I did what NSV does (This will be a theam).
- The firelocks are an intentional alteration on Kilo's normal configuration. They provide a better counter for fire and high heat and don't crush you when you need to go through them. In short, they're placed that way to provide an emergency "airlock." It also makes firelock crushing easier for mean AI's, so it's not all positive.
- Why will maint APCs act weird with the in-line wiring?
- Do you have any suggestions for relocating the grav generator?
- The empty gaps above the bar either look down on space, or on the bar. 
- the window north of tcomms is in engineering-restricted maintenance which is, if I remember correctly, only accessible from either engi or space. So it's a struggle either way.
- the invalid icon cables are vertical nodes, which haven't had a sprite generated. 



TO DO
- [x] move viro disposals a bit. (looping z makes this a bit pedantic but you have a fair point.
- [x] Make more assistant spawns around the map. (or let them brawl.)
- [x] Figure out a way to make maint south of SM less 3x3y
- [x] trim back random maint in upper z
- [x] merge from mastere
- [x] add more stairs in somehow?
- [x] Test mechs going up/down stairs.

PestoVerde322 commented:
> Why will maint APCs act weird with the in-line wiring?

It's something i've experimented on the early version of fland (pre-release) and i noticed that the rats would chew on the main cable making the APC however practically "invulnerable" to rat attacks, as one side is always guaranteed to get power (plus engineers don't really like that configuration so).

> Do you have any suggestions for relocating the grav generator?

Either on the second floor's hallway or in place of the pod next to the science shuttle.

The rest I guess then it's good to go

SergeiKoralev committed:
viro disposals outlet, adds grey spawns, fixes 3x3 hall, uppr z maint

upper z maint fixed

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
adds a stairwell and tested mechs moving up and down. (They work)

SergeiKoralev commented:
> It's something i've experimented on the early version of fland (pre-release) and i noticed that the rats would chew on the main cable making the APC however practically "invulnerable" to rat attacks, as one side is always guaranteed to get power (plus engineers don't really like that configuration so).

I'm going to categorize this as not an issue. The mouse might target the wire going under first, but a second would still get the node and there's still the added damage of no power going through next. 



> Either on the second floor's hallway or in place of the pod next to the science shuttle.
> 
> The rest I guess then it's good to go

I think I'm going to leave the grav gen in place. For most people, including nukies, it's pretty annoying and more a target of convenience. I'm not that concerned. If it winds up being a major issue I'll alter it.

PowerfulBacon commented on code/game/machinery/lightswitch.dm:
```diff
@@ -10,6 +10,23 @@
+	pixel_y = -7
+
+/obj/machinery/light_switch/Initialize()
 /obj/machinery/light_switch/Initialize(mapload)
```
🤔

SergeiKoralev replied:
That must've changed when I made directional versions of light switches. I'll remove that line shortly.

PowerfulBacon reviewed (commented):
The top floor and bottom are 2 entirely different map files? Shouldn't they be a single map file and use the z value in that map file, so the mapping tools display properly?

PowerfulBacon commented:
I definitely can feel this map having the same problems as multi-corg, the main floor is on the bottom with the top being mostly maintenance, allowing for threats that you can't see.
The maints floor should be underneath the floor with all the stuff on it.
It feels like a single-z map with an extra floor added on top, since almost everything is on 1 floor.

There is a ladder that just goes straight into security
![image](https://user-images.githubusercontent.com/26465327/176901075-ffd23190-b196-4add-9aa9-e57ee268af1f.png)

The map feels a little unpolished, it seems to lack detail in areas. For example, this hallway has no decals and feels a little lacking
![image](https://user-images.githubusercontent.com/26465327/176901458-0bfc5ffc-0fa3-4068-8ed4-871e23da864b.png)

This staircase leads straight into maintenance, why would a staircase go to maints.
![image](https://user-images.githubusercontent.com/26465327/176901542-212e4dc6-77f9-4e2a-9b6f-d84ce799d4f3.png)

Some of the random maints are made wrong, the doors should be in the middle.
![image](https://user-images.githubusercontent.com/26465327/176901625-79abb178-1204-4457-b0ce-ffd7856a8ed1.png)

There are also some missing doors.
![image](https://user-images.githubusercontent.com/26465327/176901683-ff743949-519c-4ced-8940-54ef575f2241.png)

This maintenance dead-end is a little annoying to get stuck in
![image](https://user-images.githubusercontent.com/26465327/176901736-197ee80a-9b8b-4b96-88b6-ccbd15ebc06d.png)

Medical has no cryo tubes.
Seems there are no cryo pods either.

The armoury is absolutely huge.

My main critisism is the top floor being poorly utilised, being entirely maintenance. The problem with the main floor not being the top floor is that you face threats that are completely impossible to fight since they are invisible to you.
I think in order to be a good map, a good first step would be to swap the top and bottom floors around.

PowerfulBacon commented:
Detective has no office

ghost commented:
I don't think this map has ~~any~~ enough cryopods, I can't find any in-game besides the one in perma and I don't see any more on the screenshots.

Miliviu commented:
From what I have seen the emergency maintenance access does not work on the secondary floor!

PowerfulBacon commented:
After playing blob on this map, I can say for sure there should not be an entire floor of maints that is almost impossible to access apart from some easilly destroyable ladders.

Pirill commented:
Service lathe can be accessed from maints on the top floor via ladder

Weird SM mapping, I'm not even sure those connect right;
![image](https://user-images.githubusercontent.com/29107291/176907072-5955d588-034b-4029-bfb3-6519e0218a31.png)

Weird powernet issues and confusing wiring scheme. The use of yellow and red wires seems a bit mixed up in Engineering. Also atmos wasn't even powered at the start for some reason until I exposed some wires via floor tiles. Not sure if this was a weird coincidence or a bug;

Engineering closets are locked inside secure storage, making it inconsistent with other maps, especially since secure storage has Engi access here, not CE.

Also no rapid cable layer in Engineering unless I missed it.

Weird stair ladder combo near dorms
![image](https://user-images.githubusercontent.com/29107291/176906962-10dfc7a8-b819-4ec1-b15a-b196cb7748a1.png)

Dorms could use a washing machine;

EVA has no RCD;

Upper level overall just needs more polish. It's a mass of maint rooms with little to no landmarking.

SergeiKoralev committed:
patch 0.1

SergeiKoralev committed:
Removing Bacon's light line

SergeiKoralev committed:
tweaks based on some feedback.

SergeiKoralev committed:
flips the map upside down.

SergeiKoralev committed:
I think I hunted down all the broken random maint rooms.

SergeiKoralev committed:
Pigeons need bread (or fixes) too.

SergeiKoralev committed:
more bird food

SergeiKoralev committed:
Making basement areas.

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
Fixing type paths. Hey, I'm not dead!

SergeiKoralev committed:
reviro virology

Includes relaxed (doctor) access to quarantine ward, quarantine rooms for both hemorrhagic diseases, and more benign diseases. Includes decontamination showers and hazmat suits at multiple stages before entering the infectious diseases area.

SergeiKoralev committed:
Updates viro again

PowerfulBacon reviewed (dismissed):
Approving for testmerge

SergeiKoralev committed:
bringing up to master

SergeiKoralev committed:
Fixes all the newscaster placement. Finishes disposals.

SergeiKoralev committed:
fixing linters

SergeiKoralev committed:
removing a double lattice

SergeiKoralev committed:
removing double firelock

SergeiKoralev committed:
inverted conveyors seem redundant...

SergeiKoralev committed:
PigeonFixes

SergeiKoralev committed:
Fixes a few extra bugs.

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
fixes type path miscompares.

SergeiKoralev committed:
map aggregator thinggy REEEEEEEEEE

SergeiKoralev committed:
I have no idea how that even got here.

SergeiKoralev committed:
various bug fixes

SergeiKoralev committed:
Merge remote-tracking branch 'BeeStation/master' into DeadStation

SergeiKoralev committed:
more bug patches.

SergeiKoralev committed:
working on fixing linters

SergeiKoralev committed:
more work on checks

SergeiKoralev committed:
found the airlock causing check issues.

itsmeow committed:
Merge remote-tracking branch 'origin/master' into DeadStation

itsmeow committed:
Disable multiz power adapter

itsmeow committed:
Replace gene headset

SergeiKoralev committed:
Explo airlock airlock and access fix

SergeiKoralev committed:
stairs to AI core

SergeiKoralev committed:
Another major pass of fixes.

SergeiKoralev committed:
shuttle dock and east maint

SergeiKoralev committed:
toilet and mirror

SergeiKoralev committed:
theater

SergeiKoralev committed:
repaint filter loop

SergeiKoralev committed:
openspace upper deck

SergeiKoralev committed:
trimlines in the halls
